slot.go

```diff
@@ -75,6 +77,43 @@ func (s *Slot) Key(id string) (*Key, error) {
 	})
 }
 
+// GetIDByLabel returns a hex-encoded ID for the given label
+func (s *Slot) GetIDByLabel(label string) (string, error) {
+	h, err := s.alloc()
+	if err != nil {
+		return "", err
+	}
+	defer s.release(h)
+
+	search := []*pkcs11.Attribute{
+		pkcs11.NewAttribute(pkcs11.CKA_LABEL, label),
+	}
+
+	if err := s.module.ctx.FindObjectsInit(h, search); err != nil {
+		return "", err
```

Comment:
I think it'd probably make sense to annotate a few of these bare error returns with `fmt.Errorf` to make debugging a bit easier in the future